pkg/jobrunaggregator/jobrunhistoricaldataanalyzer/analyzer.go

```diff
@@ -0,0 +1,211 @@
+	if err != nil {
+		return err
+	}
+	if len(currentHistoricalData) == 0 {
+		return fmt.Errorf("current historical data is empty, can not compare")
+	}
+
+	switch {
+	case o.newFile == "" && o.dataType == "alerts":
+		newHistoricalData, err = o.ciDataClient.ListAlertHistoricalData(ctx)
+		if err != nil {
+			return err
+		}
+	case o.newFile == "" && o.dataType == "disruptions":
+		newHistoricalData, err = o.ciDataClient.ListDisruptionHistoricalData(ctx)
+		if err != nil {
+			return err
+		}
+	default:
+		newHistoricalData, err = readHistoricalDataFile(o.newFile)
+		if err != nil {
+			return err
+		}
+	}
+
+	if len(newHistoricalData) == 0 {
+		return fmt.Errorf("new historical data is empty, can not compare")
+	}
+
+	// We check to make sure the current data set doesn't contain the previous release version
+	// If that's the case, then we are in a new release event (i.e. a new branch has been cut)
+	// We then write to a `require_review` file to record why a review would be required.
+	currentVersionUpdate := currentDataContainsPreviousRelease(previousRelease, currentHistoricalData)
+	if currentVersionUpdate {
+		if err := requireReviewFile("The current data contains previous release version"); err != nil {
+			return err
+		}
+	}
+
+	// We convert our query data to maps to make it easier to handle
+	newDataMap := convertToMap(newHistoricalData)
+	currentDataMap := convertToMap(currentHistoricalData)
+
+	result := o.compareAndUpdate(newDataMap, currentDataMap, currentRelease, currentVersionUpdate)
+
+	err = o.renderResultFiles(result)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("successfully compared (%s) with specified leeway of %s\n", o.dataType, o.leeway)
+	return nil
+}
+
+func (o *JobRunHistoricalDataAnalyzerOptions) compareAndUpdate(newData, currentData map[string]jobrunaggregatorapi.HistoricalDataRow, release string, newReleaseEvent bool) compareResults {
+	// If we're in a new release event, we don't care about the current data
+	if newReleaseEvent {
+		currentData = make(map[string]jobrunaggregatorapi.HistoricalDataRow)
+	}
+	increaseCount := 0
+	decreaseCount := 0
+	results := []parsedJobData{}
+	added := []string{}
+	for key, new := range newData {
+
+		// We only care about the current active release data, we skip all others
+		if new.Release != release {
+			continue
+		}
+
+		newP99 := getDurationFromString(new.P99)
+		newP95 := getDurationFromString(new.P95)
+		d := parsedJobData{}
+
+		// If the current data contains the new data, check and record the time diff
+		if old, ok := currentData[key]; ok {
+			oldP99 := getDurationFromString(old.P99)
+
+			d.HistoricalDataRow = new
+			d.DurationP99 = newP99
+			d.DurationP95 = newP95
+
+			timeDiff := newP99 - oldP99
+			hasPrevData := oldP99 != 0
+			if newP99 > oldP99 && timeDiff > o.leeway && hasPrevData {
+				increaseCount += 1
+				d.TimeDiff = timeDiff
+			}
+			if newP99 < oldP99 {
+				decreaseCount += 1
```

Comment:
Do we care about sudden large decrease in timeDiff?

Reply:
I'm not sure, @dgoodwin do you have an opinion on weather we want to trigger a pr review if the data set goes in the opposite direction, so not only do we trigger if we increase by the leeway but we also trigger if we decrease by the leeway?

Reply:
Yes let's trigger the review, just to pat ourselves on the back if nothing else.